metagraph/src/common/elias_fano.cpp

```diff
@@ -223,6 +223,15 @@ EliasFanoEncoder<T>::~EliasFanoEncoder() {
     assert(!sink_internal_.is_open());
 }
 
+template <typename T>
+void EliasFanoEncoder<T>::append(const std::vector<T> &data,
+                                 const std::string &out_fname) {
+    std::ofstream sink(out_fname, std::ios::binary | std::ios::app);
+    std::ofstream sink_upper(out_fname + ".up", std::ios::binary | std::ios::app);
+    EliasFanoEncoder<T> encoder(data, &sink, &sink_upper);
+    encoder.finish();
```

Comment:
With the current implementation, it looks like it's only possible to append once since it finishes the encoder afterwards. This seems like an artificial limitation.

Why not call `finish` externally instead?

Reply:
This thing writes in blocks, finish flushed the buffers.
Multiple appends are supported, no issue with that.

Reply:
Maybe it would be good to have a unit test for this.

https://github.com/ratschlab/metagraph/blob/b4c7aab2820e40a2b7543e83d009b6cc45276e17/metagraph/src/common/elias_fano.cpp#L274
On this line, there's a check to make sure that all elements have been added, so wouldn't this assert false?

Reply:
Unless there's something I'm missing here...

Reply:
It just pushed another block to the file.
I don't see what confuses you. Can you explain what seems to you?

Reply:
I mean, declared_size_ and size_ for this next block will be different. The blocks are independent. They just sit back to back in the file.

Reply:
ok, my bad. for some reason I thought it was appending to an existing block, but yes, since a new object is created, I see that it's a new independent block